### PR TITLE
Move dev docker images to airflow registry

### PR DESCRIPTION
**Pull request by potiuk**
Part of #9401

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by dimberman** [comment]
This... feels off to me.

Why are we maintaining images for external libraries? If those projects and images have proper licences why would we want to take on that responsibility for ourselves?

**Comment by potiuk** [comment]
We are not taking more responsibility. We are making sure that we can reproducibly rebuild those images and have control over it by the community. Those two are different things.

 See the https://lists.apache.org/thread.html/r0d0f6f5b3880984f616d703f2abcdef98ac13a070c4550140dcfcacf%40%3Cdev.airflow.apache.org%3E thread where I explained my reasoning and worries. 

This change is really for making sure we have control over those images and that the users of those images have trackable origin of the images (in this solution it is trackable down to the commit). If the original image/library changes, we can migrate to use the new one.

It's not a big issue In most cases for "dev" dependencies - we would maybe touch it once a year (or when there is a concrete need to upgrade). So I just added it for consistency.

But it's a different story for the "runtime" dependencies in the helm chart. I believe that when we release the helm chart to our users, it is already our responsibility to make sure this image is patched with latest security fixes from "official" sources. Simply because by releasing those, we are suggesting our users to use those images - just by providing them officially released helm chart. I'd complain as a user if someone released a chart for production use and told me to use it when at the moment of release latest available patches are not applied. I think it's simply a consequence of releasing a production-ready chart to our users and the responsibility for that is already on us. As a PMC, this is responsibility I take when I do "+1" for such release.  I feel this is my responsibility as well, because as far as I know it has not been (yet) in any way standardized by ASF. I started a thread here: https://lists.apache.org/thread.html/rf2af2a95e7687fe94ede23fe9df388f784c8231a5968b109f677cbe8%40%3Cbuilds.apache.org%3E but there are different approaches different projects do it seems. So we have a freedom of deciding how we are going to approach it ourselves. 

Again - this chart is intended for production use, so IMHO - with every "official release" we should at the very least rebuild those images using the latest version of the base images (containing latest security patches) and the latest version of the "tools" we are releasing (for example pgbouncer).  Obviously - we cannot be responsible to upgrade those in-between releases, but "official" release by the community simply should not use an unpatched based image version as dependency - especially that it is easy to do.

By having a simple way of rebuilding the images by anyone and publishing them in our registry, we do not "increase" our responsibility - the responsibility is already there. We are increasing community control over it instead.


I think this is not at all difficult, burdensome and problematic to use this approach. It's easy to follow, reproducible and IMHO does not oblige us to anything more than needed.

**Comment by potiuk** [comment]
Actually I found that:

https://www.apache.org/legal/release-policy.html#what

> The Apache Software Foundation produces open source software. All releases are in the form of the source materials needed to make changes to the software being released. In some cases, binary/bytecode packages are also produced as a convenience to users that might not have the appropriate tools to build a compiled version of the source. In all such cases, the binary/bytecode package must have the same version number as the source release and may only add binary/bytecode files that are the result of compiling that version of the source code release.

**Comment by potiuk** [comment]
Hey @dimberman @schnie - > I would love a review for it (see email)

**Comment by potiuk** [comment]
Hey everyone - after merging the "helm" images in, I also rebase the old PR of mine where I brought sources for all the images we are using for DEV. Those are not strictly needed for any user-facing stuff, it's more for us - that in case (for any reason) we want to replace any of the images, we know where to start. I know for example @mik-laj is working on an improved Kerberos image (he is adding Kerberos capability to presto operators and we want to make sure that we can test it in our integration tests pipeline with Kerberos server that actually works. That is just an example - we might want to upgrade any of our deps at any time, and having the sources would be rather useful IMHO.

**Comment by potiuk** [comment]
Seems I fixed one error - it should be green now!

**Comment by potiuk** [comment]
Updated to the Google Shell Giuide. I would love to merge this one, this way we completely close image case - all under our control, with easy way to rebuild by the users.

**Comment by ashb** [comment]
This will be doubly useful cos of https://www.docker.com/pricing/resource-consumption-updates

**Comment by potiuk** [comment]
> This will be doubly useful 

What do you mean @ashb ?

**Comment by ashb** [comment]
With docker's change in policies, "inactive" images run the risk of being deleted.

(Unlikely as it resets the timer on any pull, but this is a risk for the "random" images if we use the upstream/stock ones)

**Review by kaxil — APPROVED** [review]

File: scripts/ci/dockerfiles/apache-rat/Dockerfile
Comment (kaxil) [inline scripts/ci/dockerfiles/apache-rat/Dockerfile]: Do we need to change this as a label? Saw in one of your other prs

File: scripts/ci/dockerfiles/krb5-kdc-server/supervisord.conf
Comment (kaxil) [inline scripts/ci/dockerfiles/krb5-kdc-server/supervisord.conf]: Double license on this file

File: scripts/ci/dockerfiles/stress/Dockerfile
Comment (kaxil) [inline scripts/ci/dockerfiles/stress/Dockerfile]: According to Official Docker Documentation the MAINTAINER instruction is deprecated. Instead, one should use the LABEL instruction to define the author of the generated images. The LABEL instruction is more flexible, enables setting metadata, and can be easily viewed with the command docker inspect .


Same here

**Comment by potiuk** [comment]
I have not such strong feelings about the dev images as I have for those that we share with the users, but yeah - and in this case the "pull" one is I think important, but If you are ok with having those in - i am all for it. I think this will make it really "self-sufficient" in many ways, without relying on "random images". And I am happy to revert it, if it will turn to be too much of a maintenance effort (which I seriously doubt - those images are changed once every blue moon).

File: scripts/ci/dockerfiles/stress/Dockerfile
Comment (potiuk) [reply scripts/ci/dockerfiles/stress/Dockerfile]: Yep. I am going to rebase and apply the hadolint rules from master.

File: scripts/ci/dockerfiles/apache-rat/Dockerfile
Comment (potiuk) [reply scripts/ci/dockerfiles/apache-rat/Dockerfile]: Changed.

File: scripts/ci/dockerfiles/krb5-kdc-server/supervisord.conf
Comment (potiuk) [reply scripts/ci/dockerfiles/krb5-kdc-server/supervisord.conf]: Removed

**Comment by ashb** [comment]
Oh yeah, I'm definitely happy with this. It seems like the right thing for now (and no decision is permanent if we discover something new in the future)

**Review by kaxil — APPROVED** [review]

**Pull request merged**
